sdk/core/Azure.Core/src/RequestContext.cs

```diff
@@ -52,5 +59,74 @@ public void AddPolicy(HttpPipelinePolicy policy, HttpPipelinePosition position)
             Policies ??= new();
             Policies.Add((position, policy));
         }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="statusCodes">Status codes that will not be considered to be error status codes for the scope of the service method call.</param>
+        /// <param name="isError">Whether the passed-in status codes will be considered to be error codes for the duration of this request.</param>
+        public void ConfigureResponse(int[] statusCodes, bool isError)
+        {
+            if (isError)
+            {
+                _customErrorCodes ??= new();
+                _customErrorCodes.AddRange(statusCodes);
+            }
+            else
+            {
+                _customNonErrorCodes ??= new();
+                _customNonErrorCodes.AddRange(statusCodes);
+            }
+        }
+
+        internal ResponseClassifier GetResponseClassifier(ResponseClassifier inner)
+        {
+            return new PerCallResponseClassifier(inner, _customErrorCodes, _customNonErrorCodes);
+        }
+
+        private class PerCallResponseClassifier : ResponseClassifier
+        {
+            private readonly ResponseClassifier _inner;
+            private readonly IList<int>? _customErrorCodes;
+            private readonly IList<int>? _customNonErrorCodes;
+
+            public PerCallResponseClassifier(ResponseClassifier inner, IList<int>? errorCodes, IList<int>? nonErrorCodes)
+            {
+                _inner = inner;
+                _customErrorCodes = errorCodes;
+                _customNonErrorCodes = nonErrorCodes;
+            }
+
+            public override bool IsRetriableResponse(HttpMessage message)
+            {
+                return _inner.IsRetriableResponse(message);
+            }
+
+            public override bool IsRetriableException(Exception exception)
+            {
+                return _inner.IsRetriableException(exception);
+            }
+
+            public override bool IsRetriable(HttpMessage message, Exception exception)
+            {
+                return _inner.IsRetriable(message, exception);
+            }
+
+            public override bool IsErrorResponse(HttpMessage message)
```

Comment:
The main thing that worries me is that the arrays "win" over the inner classifier. Is that OK? Will the user understand that?

Reply:
The thinking is that, if the caller sets a status code as an error on the RequestContext, it will be considered an error for the duration of the call, regardless of what the client ResponseClassifer thinks.  

The caller would write:
```
RequestContext context = new();
context.ConfigureStatus(new int[] { 404 }, isError: false);
client.Delete(id, context);  // doesn't throw on a 404, where default classifier would have said 404 was an error
```

Is there something else you think would be more expected by users?

Reply:
Per offline discussion, I'll explore adding these arrays as fields on ResponseClassifier